tests/phpunit/tests/media.php

```diff
@@ -3710,6 +3710,86 @@ function() {
 		$this->assertSame( 3, $omit_threshold );
 	}
 
+	/**
+	 * Tests that wp_get_loading_attr_default() returns the expected loading attribute value before loop but after get_header.
+	 *
+	 * @ticket 58211
+	 *
+	 * @covers ::wp_get_loading_attr_default
+	 *
+	 * @dataProvider data_wp_get_loading_attr_default_before_and_no_loop
+	 *
+	 * @param string $context
+	 */
+	public function test_wp_get_loading_attr_default_before_loop( $context ) {
```

Comment:
This test appears to be performing three separate scenarios. These should be performed in separate test methods so that each scenario is tested individually.

Reply:
@costdev This function was meant to show how the value of `wp_get_loading_attr_default` changes in this special condition. More than three single tests this is a compound test to check the flow. 

Somewhat similar to what was done in [test_wp_omit_loading_attr_threshold_filter](https://github.com/WordPress/wordpress-develop/blob/5db4787bc4b6c83351d84e4366d038ae863ae80e/tests/phpunit/tests/media.php#L3613)

Reply:
@costdev @kt-12 I'm personally not sure which of those is the way to go. For context, the "right" way to write WordPress unit tests doesn't exist to my knowledge, and different people apply vastly different patterns.

More recently, there appear to be a few new best practices to have been established, yet I am missing a place where they are documented. As long as they're not, we'll always end up with inconsistencies like this.

My personal take is that one assertion per test is preferable as long as it's "reasonable". If the logic for setting up the environment for a certain scenario is quite complicated, I feel like combining multiple assertions can be okay, as long as every assertion receives a more specific message for its failure ("Failed asserting that...).

Of course the existing tests don't have that, since they were written a while ago where no reviewer of that code paid attention to it 🙃 

But for this PR, I'd be in favor of keeping the tests the way they are, but add messages for each assertion to clarify what they are for. Let me know what you both think.

Reply:
As any one of these scenarios may fail while others pass, if they're all tested in one method, later assertions won't run. By separating these into the specific scenario being tested, we can identify the extent of any failures/regressions immediately, which might help us track down the source of the issue as well.

Reply:
While, I was favouring what @felixarntz said of having messages for each assertion but what @costdev have said is also true. If any of the initial assertions fails then we may not know what would happen with other assertion in same function. I have split the test into three different tests.

Reply:
The new format of tests looks excellent to me - @costdev can you please give this another look?

Reply:
Thanks for the updates @kt-12! These look great to me!

@felixarntz Thanks for the ping for me to follow up on this! We might consider splitting the `wp_get_loading_attr_default()` tests out to a separate file at some point, as well as taking another look at [pre-existing tests such as this one](https://github.com/wordpress/wordpress-develop/blob/02d89bc44d19ec72a036427a46804a52440f493b/tests/phpunit/tests/media.php#L3561) that might benefit from splitting up for the same reason - quite a lot of `assertSame()/assertFalse()` without messages or tests that appear to be testing multiple things.